src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProvider.cs

```diff
@@ -230,7 +230,23 @@ private async Task<IReadOnlyList<SecretListEntry>> FetchAllSecretsAsync(Cancella
 
                     var request = new GetSecretValueRequest { SecretId = secret.ARN };
                     Options.ConfigureSecretValueRequest?.Invoke(request, new SecretValueContext(secret));
-                    var secretValue = await Client.GetSecretValueAsync(request, cancellationToken).ConfigureAwait(false);
+                    GetSecretValueResponse? secretValue;
+
+                    try
+                    {
+                        secretValue = await Client.GetSecretValueAsync(request, cancellationToken).ConfigureAwait(false);
+                    }
+                    catch (ResourceNotFoundException)
```

Comment:
Did you try using `catch (Exception) when (filter)` ?

Reply:
Indeed, good catch. Updated.